docs/markdown/Wrap-dependency-system-manual.md

```diff
@@ -47,24 +47,35 @@ Currently Meson has four kinds of wraps:
 
 Wrap files are written in ini format, with a single header containing the type
 of wrap, followed by properties describing how to obtain the sources, validate
-them, and modify them if needed. An example wrap-file for the wrap named
-`libfoobar` would have a filename `libfoobar.wrap` and would look like this:
+them, and modify them if needed.
+
+Since *0.56.0* the same syntax as [machine files](Machine-files.md) is supported
+and recommended for values:
+- Quoted string values (e.g. `directory='libfoobar-1.0'`).
+- Python-style list values (e.g. `program_names=['prog1', 'prog2']`)
+- `+` and `/` operators and `[constants]` section (e.g. `source_url='http://example.com' / project_name + 'tar.gz'`)
```

Comment:
Does the `/` filename operator work as expected when used with *urls* on Windows?

And `tar.gz` should be `.tar.gz`.

Reply:
Argh, indeed it fails on Windows CI: `AssertionError: 'http://mirror.invalid\\project-1.0.tar.gz' != 'http://mirror.invalid/project-1.0.tar.gz'`.

Reply:
It is actually a bug in machine files too, '/' operator was never supposed to add '\'. interpreterbase.py does that for the / operator in meson.build files: `os.path.join(*args).replace('\\', '/')`.

Fixed both.

Reply:
Actually replaced in both places using posixpath.join() since that will preserve any existing `\` char in the path.

Reply:
It's not clear to me that there's a benefit for urls vs. just hardcoding the `/` character and using `+`. Paths are logically justified specifically and solely because cross-platform handling has nuances above and beyond "just do string concatenation".

IMHO, YMMV.

Reply:
It has the cosmetic advantage that `'foo/' / 'bar'` and `'foo' / 'bar'` both yields the same string `'foo/bar'`, avoiding the double slash.

Reply:
and when used with variables, `foo / bar` is nicer than `foo + '/' + bar`

Reply:
(unsolved this thread because it's unclear if we can use posixpath.join() here)

Reply:
Ok, reverted to use `os.path.join(l, r).replace('\\', '/')`. It is of course not fool proof for URLs but works fine in simple cases.